bokehjs/src/coffee/models/renderers/graph_renderer.coffee

```diff
@@ -66,11 +70,48 @@ export class GraphRenderer extends Renderer
       selector = @node_renderer.data_source.selection_manager.selector
       selector.update(indices, final, append)
       @node_renderer.data_source.selected = selector.indices
+
+      if @selection_mode == "linked"
+        node_indices = (@node_renderer.data_source.data.index[i] for i in indices["1d"].indices)
+        edge_source = @edge_renderer.data_source
+        edge_indices = []
+        for i in [0...edge_source.data.start.length]
+          if contains(node_indices, edge_source.data.start[i]) or contains(node_indices, edge_source.data.end[i])
+            edge_indices.push(i)
+
+        linked_index = create_hit_test_result()
+        for i in edge_indices
+          linked_index["2d"].indices[i] = [0]
```

Comment:
Selecting the entire array is tough because it's coordinates are supplied by the layout_provider. Currently we only support 2-element multilines (i.e. one segment), so this selects the whole segment. This will probably need to be readdressed in the future,